contracts/Lottery.sol

```diff
@@ -19,72 +20,103 @@ contract Lottery {
     struct LotteryTicket {
         uint lotteryId;
         uint price;
-        address payable ticketOwner;
+        address ticketOwner;
     }
     
 
     address private owner;
-    uint32 private totalAmount;
-    uint32 private ticketPrice;
-    Player[] public winner;
+    LotteryState private state;
+    uint private ticketPrice;
+    Player private winner;
     
-    mapping (address => Player) playerMap;
     Player[] players;
     LotteryTicket[] tickets;
-
+    
+    mapping (address => Player) playerMap;
+    mapping(address => LotteryTicket[]) playerLotteryTicketsMapping;
+    
+    //events
+    event NewUser(address user);
+    event BuyTicket(address buyer, uint ticketsCount, uint totalAnount);
+    
+    event LotteryWinner(address winner, uint amount);
+    event LotteryStatusUpdate(address player, string status);
+    event LotteryStatusUpdateWithMessage(address player, string status, string message);
+    
+    // This will be used if lottery owner does premature termination of lottery and in this case amount should be refunded to each player
+    event RefundLottery(address player, uint amount); 
+    
     constructor () public {
         state = LotteryState.NotStarted;
         owner = msg.sender;
     }
-
-    function registerPlayer(string memory _name, string memory _emailId, string memory  _phoneNumber) public payable returns (string memory result) {
+    
+    function registerPlayer(string memory _name, string memory _emailId, string memory  _phoneNumber) public {
         //Owner shouldn't be player
-        require(owner != msg.sender);
-        require((state != LotteryState.NotStarted) && (state != LotteryState.Finished), "Player can't register while lottery has started or finished.");
+        require(owner != msg.sender, "Lottery owner can't register as player.");
+        require(state == LotteryState.Started, "Player can't register while lottery has started or finished.");
```

Comment:
Why do we need to restrict user registrations if the lottery has started?

Reply:
Corrected, this was initial condition...